flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/datatransfer/DataTransferStrategyBuilder.java

```diff
@@ -45,20 +45,25 @@ public class DataTransferStrategyBuilder {
     private static final Logger LOG = LoggerFactory.getLogger(DataTransferStrategyBuilder.class);
 
     public static DataTransferStrategy buildForSnapshot(
+            SnapshotType.SharingFilesStrategy sharingFilesStrategy,
             @Nullable ForStFlinkFileSystem forStFlinkFileSystem,
             @Nullable CheckpointStreamFactory checkpointStreamFactory) {
         return buildForSnapshot(
+                sharingFilesStrategy,
                 forStFlinkFileSystem,
                 isDbPathUnderCheckpointPathForSnapshot(
                         forStFlinkFileSystem, checkpointStreamFactory));
     }
 
     @VisibleForTesting
     static DataTransferStrategy buildForSnapshot(
+            SnapshotType.SharingFilesStrategy sharingFilesStrategy,
             @Nullable ForStFlinkFileSystem forStFlinkFileSystem,
             boolean isDbPathUnderCheckpointPathForSnapshot) {
         DataTransferStrategy strategy;
-        if (forStFlinkFileSystem == null || isDbPathUnderCheckpointPathForSnapshot) {
+        if (sharingFilesStrategy != SnapshotType.SharingFilesStrategy.FORWARD_BACKWARD
```

Comment:
I think for `ForStIncrementalSnapshotStrategy`, the `SnapshotType.SharingFilesStrategy.FORWARD` is not supported since it breaks the precondition of file sharing between CP and DB.